src/transforms/filter.jl

```diff
@@ -69,22 +65,55 @@ DropMissing(cols::T...) where {T<:ColSelector} =
 
 isrevertible(::Type{<:DropMissing}) = true
 
-_ftrans(::DropMissing{Colon}, table) =
+# ftrans
+_ftrans(::DropMissing{Colon}, cols) =
   Filter(row -> all(!ismissing, row))
 
-function _ftrans(transform::DropMissing, table)
-  allcols = Tables.columnnames(table)
-  cols = _filter(transform.colspec, allcols)
+_ftrans(::DropMissing, cols) =
   Filter(row -> all(!ismissing, getindex.(Ref(row), cols)))
+
+# nonmissing 
+_nonmissing(::Type{T}, c) where {T} = c
+_nonmissing(::Type{Union{Missing,T}}, c) where {T} =
+  collect(T, c)
+
+function _nonmissing(columns, col)
+  c = Tables.getcolumn(columns, col)
+  _nonmissing(eltype(c), c)
+end
+
+function _nonmissing(table, cols, allcols)
```

Comment:
I think this is an overcomplicated design. Can we simply pass the table as the first argument and retrieve the column names inside the function instead of passing allcols? Also, try to avoid creating external functions unless that is strictly necessary. I thought that we should be able to write a compact apply and revert doing pre and post processing without all this additional helper functions.

Reply:
OK